test/unit/test_disco_rds.py

```diff
@@ -298,3 +298,105 @@ def test_get_rds_security_group_id(self):
         sg_group_id = self.rds.get_rds_security_group_id()
 
         self.assertEqual(MOCK_SG_GROUP_ID, sg_group_id)
+
+    def test_delete_db_instance_with_snapshot(self):
+        """Test delete db instance with snapshot"""
+        self.rds.client.describe_db_instances.return_value = {
+            'DBInstances': [{
+                'DBInstanceIdentifier': 'unittestenv-db-name',
+                'AllocatedStorage': '500GB',
+                'DBSubnetGroup': {
+                    'DBSubnetGroupName': 'group_name'
+                }
+            }]
+        }
+        with patch('__builtin__.raw_input', return_value='500GB'):
+            self.rds.delete_db_instance("unittestenv-db-name", True)
+            self.rds.client.delete_db_subnet_group.assert_called_with(DBSubnetGroupName="group_name")
+            self.rds.client.delete_db_instance.assert_called_with(DBInstanceIdentifier="unittestenv-db-name",
+                                                                  SkipFinalSnapshot=True)
+
+    def test_delete_db_instance_err_size(self):
+        """Test delete db instance with snapshot but incorrect size provided"""
+        self.rds.client.describe_db_instances.return_value = {
+            'DBInstances': [{
+                'DBInstanceIdentifier': 'unittestenv-db-name',
+                'AllocatedStorage': '500GB',
+                'DBSubnetGroup': {
+                    'DBSubnetGroupName': 'group_name'
+                }
+            }]
+        }
+        with patch('__builtin__.raw_input', return_value='100GB'):
+            self.assertRaises(SystemExit, self.rds.delete_db_instance, "unittestenv-db-name", True)
+            assert not self.rds.client.delete_db_subnet_group.called
```

Comment:
For consistency, we usually do `self.assertFalse(self.rds.client.delete_db_subnet_group.called)`